### PR TITLE
EDU-1001 Generalize brute force defence over all respective endpoints

### PR DESCRIPTION
**Pull request by ahelmberger**
Closes https://educandu.atlassian.net/browse/EDU-1001

### CODE REVIEW

File: src/server/user-controller.js [hunk 1]
```diff
@@ -222,31 +220,20 @@ class UserController {
...
   }
 
-  async handlePostUserLogin(req, res, next) {
-    const failedAttemptsCount = await this.requestLimitRecordService.getCount({ req });
-    if (failedAttemptsCount >= FAILED_LOGIN_ATTEMPTS_LIMIT) {
-      throw new TooManyRequests();
-    }
-
-    passport.authenticate('local', async (err, user) => {
+  handlePostUserLogin(req, res, next) {
```
Comment (ahelmberger) [inline src/server/user-controller.js:223]: This function is basically reset to the state before https://educandu.atlassian.net/browse/EDU-976

File: src/server/user-controller.js [hunk 2]
```diff
@@ -98,9 +96,9 @@ class UserController {
     this.requestLimitRecordService = requestLimitRecordService;
     this.passwordResetRequestService = passwordResetRequestService;
 
-    this.passwordResetRateLimitOptions = {
-      maxRequests: PASSWORD_RESET_REQUEST_LIMIT,
-      expiresInMs: PASSWORD_RESET_REQUEST_TIME_WINDOW_IN_MS,
+    this.antiBruteForceRequestLimitOptions = {
+      maxRequests: ANTI_BRUTE_FORCE_MAX_REQUESTS,
+      expiresInMs: ANTI_BRUTE_FORCE_EXPIRES_IN_MS,
       service: this.requestLimitRecordService
```
Comment (adrianaluchian) [inline src/server/user-controller.js:102]: Actually is the `requestLimitRecordService` used anymore in this case? 🤔

Comment (ahelmberger) [reply src/server/user-controller.js:102]: Yes, it is used by the middleware that is configured using these options.

Comment (ahelmberger) [reply src/server/user-controller.js:102]: BUUT, good catch: I can delete 2 of the 3 service methods! Will do.

Comment (ahelmberger) [reply src/server/user-controller.js:102]: Done!

**Review by adrianaluchian — APPROVED** [review]

**Pull request merged**